### PR TITLE
[bug] update products #34

### PR DESCRIPTION
**Pull request by Taiwola**
Update Product
​
## Description

This endpoint allows updating the details of a product item. It accepts various product attributes and updates the corresponding fields in the database, including the updated_at timestamp to reflect the time of the update.
​
## Related Issue (Link to Github issue)

The use of make in the product factory resulted in the creation of a user instance that was not persisted in the database. This caused problems during the seeding process where product creation failed due to missing user references.

### **Steps to Reproduce**
Run the database seeder to seed the products.
Observe the foreign key constraint violation error related to the user reference.

### **Solution**
The fix involves replacing the make function with the create function in the product factory to ensure that the user instance is properly persisted in the database.

Update product issue: https://github.com/hngprojects/hng_boilerplate_nestjs/issues/188
​
## How Has This Been Tested?

I used Thunder Client to test each endpoint to ensure that the code works as expected without any issues.

Testing Environment:

> Local development server
> Thunder Client extension in Visual Studio Code

### **Tests Run**

1. Update Product with Valid Data:

-               Sent a PUT request to /api/v1/products/{id} with valid product data.
-               Verified that the product was updated successfully and received a 200 OK response with the updated product details.

2. Update Product with Missing Fields:

-                 Sent a PUT request to /api/v1/products/{id} with only some of the fields provided.
-                     Verified that only the provided fields were updated and received a 200 OK response with the updated product details.

3. Update Non-Existent Product:

-                  Sent a PUT request to /api/v1/products/nonexistent-id to test the response for a non-existent product.
-                  Verified that the response was a 404 Not Found with an appropriate error message.

4, Update Product with Missing Authentication:

-            Sent a PUT request to /api/v1/products/{id} without authentication headers.

-                 Verified that the response was a 401 Unauthorized.

​
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
​
## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by Taiwola** [comment]
@onyx093 
@Goldengide 
@tulbadex 
@bhimbho 

pls review

**Comment by Taiwola** [comment]
@Dev-Tonia 
@Muhammad235

**Review by Dev-Tonia — APPROVED** [review]
You fixed a bug not a new feature,
Rename to fix

**Review by Dev-Tonia — COMMENTED** [review]
Rename to bug fix

**Comment by timiajayi** [comment]
add link to github issue
the original issue and the one youre updating

**Comment by Taiwola** [comment]
okk.....

**Comment by Taiwola** [comment]
i have updated it @timiajayi

**Comment by Taiwola** [comment]
@Ruthiejayjay

**Comment by Taiwola** [comment]
@Ruthiejayjay

**Comment by amowogbaje** [comment]
> @onyx093 @Goldengide @tulbadex @bhimbho
> 
> pls review

Bro, I don't have the power to review I noticed you have some pulling and merge conflicts to solve though

File: tests/Feature/BlogSearchTest.php
Comment (bhimbho) [inline tests/Feature/BlogSearchTest.php:1]: why is this test file removed? this is a test written by @timiajayi for his work done on Blog search endpoint

**Review by bhimbho — CHANGES_REQUESTED** [review]

File: tests/Feature/BlogSearchTest.php
Comment (bhimbho) [reply tests/Feature/BlogSearchTest.php:1]: also merge conflicts

**Pull request closed without merging**